functions/ocr/app/detect.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2018, Google, LLC.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_ocr_detect]
+
+package ocr
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"cloud.google.com/go/pubsub"
+	"golang.org/x/text/language"
+	visionpb "google.golang.org/genproto/googleapis/cloud/vision/v1"
+)
+
+// detectText detects the text in an image using the Google Vision API.
+func detectText(ctx context.Context, bucketName, fileName string) error {
+	log.Printf("Looking for text in image %v", fileName)
+	maxResults := 1
+	image := &visionpb.Image{
+		Source: &visionpb.ImageSource{
+			GcsImageUri: fmt.Sprintf("gs://%s/%s", bucketName, fileName),
+		},
+	}
+	annotations, err := visionClient.DetectTexts(ctx, image, &visionpb.ImageContext{}, maxResults)
+	if err != nil {
+		return fmt.Errorf("DetectTexts: %v", err)
+	}
+	text := ""
+	if len(annotations) > 0 {
+		text = annotations[0].Description
+	}
+	if len(annotations) == 0 || len(text) == 0 {
+		log.Printf("No text detected in image %q. Returning early.", fileName)
+		return nil
+	}
+	log.Printf("Extracted text %q from image (%d chars).", text, len(text))
+
+	detectResponse, err := translateClient.DetectLanguage(ctx, []string{text})
```

Comment:
Why do we need to detect the source language? The Translate API can auto-detect the source language when translating, so this API call is not needed.

Reply:
It's in the other samples, I think if it detects if the text is already in the target language it uploads it to the result bucket without calling the translate cloud function

Reply:
Also I noticed the TRANSLATE bool in config isn't checked anywhere in the other samples, should I leave it out here as well?

Reply:
Ahh. That makes sense. Yeah, I would delete it from the config if it's not referenced anywhere.

cc @grant